pkg/asset/agent/manifests/common.go

```diff
@@ -2,51 +2,43 @@ package manifests
 
 import (
 	"github.com/openshift/installer/pkg/asset/agent"
-	"github.com/openshift/installer/pkg/asset/agent/agentconfig"
 	"github.com/openshift/installer/pkg/types"
 	"github.com/openshift/installer/pkg/version"
 )
 
 func getAgentClusterInstallName(ic *agent.OptionalInstallConfig) string {
-	return ic.Config.ObjectMeta.Name
+	return ic.ClusterName()
 }
 
 func getClusterDeploymentName(ic *agent.OptionalInstallConfig) string {
-	return ic.Config.ObjectMeta.Name
+	return ic.ClusterName()
 }
 
 func getInfraEnvName(ic *agent.OptionalInstallConfig) string {
-	return ic.Config.ObjectMeta.Name
+	return ic.ClusterName()
 }
 
 func getPullSecretName(ic *agent.OptionalInstallConfig) string {
-	return ic.Config.ObjectMeta.Name + "-pull-secret"
+	return ic.ClusterName() + "-pull-secret"
 }
 
 func getObjectMetaNamespace(ic *agent.OptionalInstallConfig) string {
-	return ic.Config.Namespace
-}
-
-func getNMStateConfigName(a *agentconfig.AgentConfig) string {
-	return a.Config.ObjectMeta.Name
+	if ic.Config != nil {
+		return ic.Config.Namespace
```

Comment:
Why is it using namespace instead of ic.ClusterName()

Reply:
We still need to come up with an appropriate way to generate the namespace for ZTP manifests to go in. I had the thought that maybe we should generate a kustomize file to allow the user to override it for everything in one place. In any event, up to now we've used the namespace from the InstallConfig (which most people don't usually set), and this PR doesn't change it.